hub/api/dataset.py

```diff
@@ -32,32 +37,36 @@ def __init__(
             mode (str): Mode in which the dataset is opened.
                 Supported modes include ("r", "w", "a") plus an optional "+" suffix.
                 Defaults to "a".
-            ds_slice (slice): The slice object restricting the view
-                of this dataset's tensors. Defaults to slice(None, None, None).
-                Used internally for iteration.
-            provider (StorageProvider, optional): The storage provider used to access
-                the data stored by this dataset.
+            ds_slice (slice): The slice object restricting the view of this dataset's tensors.
+                Defaults to slice(None, None, None). Used internally for iteration.
+            memory_cache_size (int): The size of the memory cache to be used in MB.
+            local_cache_size (int): The size of the local filesystem cache to be used in MB.
+            storage (StorageProvider, optional): The storage provider used to access
+            the data stored by this dataset. If this is specified, the path given is ignored.
+
 
         Raises:
             ValueError: If an existing local path is given, it must be a directory.
-            UserWarning: Both a path and provider should not be given.
+            UserWarning: Both a path and storage should not be given.
         """
         self.mode = mode
         self.slice = ds_slice
 
-        if provider is not None and path:
+        if storage is not None and path:
             warnings.warn(
-                "Dataset should not be constructed with both provider and path."
+                "Dataset should not be constructed with both storage and path. Ignoring path and using storage."
```

Comment:
good idea adding the last part "Ignoring path and using storage." This is extremely helpful actually!